### ISSUE TITLE
Feat: Hide the tabline conditionally

### ISSUE THREAD

**Issue by snorkysnark** [nvim-lualine/lualine.nvim#395]
### Requested feature
In global options, add a function that is called on redraw to determine whether the tabline should be visible. For example, you can only show it if there is more than one tab/more than one buffer, etc.
```lua
require('lualine').setup{
  tabline = {
    ...
  },
  tabline_visible = function()
    return vim.fn.tabpagenr('$') > 1
  end
}
```

### Motivation

Currently, this plugin forces the tabline to be always visible, decreasing the available screen space. Since the option is set during the first redraw, I haven't found a good way to override it with autocmds.
As people have different tabline configurations (with buffers and other custom components), you need a more flexible alternative to vim's showtabline=1 (that shows and hides the tabline based on tab count)

**Comment by shadmansaleh** [comment]
For now I'm dropping this idea. As it's too complicated to add for too little gain for now.

**Comment by amerlyq** [comment]
I wanted to achieve the same effect as airline -- to hide tabline when there is only single buffer:
```
let g:airline#extensions#tabline#buffer_min_count = 2
```
But I don't understand how to use your workaround above to achieve the same thing.
@shadmansaleh do you have any other suggestion?

Maybe simple corner case of "hide when there is only one buffer/tab" can be much more easily hacked, 
than the proper solution of "evaluate conditional function inside of tabline" ?

**Comment by AlxHnr** [comment]
This hack does not work in neovim 0.7.0:

> ```viml
> au OptionSet showtabline :set showtabline=1
> ```

For some reason OptionSet is not fired when lualine does it. I can set `showtabline=1` manually, but when switching colorschemes it reappears. Is there any reason why we can't add an option to simply leave `showtabline` alone [here](https://github.com/nvim-lualine/lualine.nvim/blob/dfac736a015f1c295eae517d9af0db87413a88ac/lua/lualine.lua#L256-L264)?

**Comment by mario-amazing** [comment]
@shadmansaleh, because of not working OptionSet in neovim 0.7.0, can we reopen this feature request?

**Comment by matheussampaio** [comment]
Maybe an easier approach is to let users control if we want [Lualine to set `showtabline=2`](https://github.com/nvim-lualine/lualine.nvim/blob/3cf45404d4ab5e3b5da283877f57b676cb78d41d/lua/lualine.lua#L508)?

If we wrap that in an condition, it would allow users to set `showtabline=1` in their config and get the default behavior of only displaying the tabline if at least two tabs exists:

```
'showtabline' 'stal'	number	(default 1)
			global
	The value of this option specifies when the line with tab page labels
	will be displayed:
		0: never
		1: only if there are at least two tab pages
		2: always
	This is both for the GUI and non-GUI implementation of the tab pages
	line.
	See |tab-page| for more information about tab pages.
```

How #736 supports hiding the tabline conditionally? I couldn't understand from read the code.

**Comment by isaif** [comment]
> Maybe an easier approach is to let users control if we want [Lualine to set `showtabline=2`](https://github.com/nvim-lualine/lualine.nvim/blob/3cf45404d4ab5e3b5da283877f57b676cb78d41d/lua/lualine.lua#L508)?
> 
> If we wrap that in an condition, it would allow users to set `showtabline=1` in their config and get the default behavior of only displaying the tabline if at least two tabs exists:

I agree that this will be very useful as we can use whatever options vim provides by default.
If this worked then we could also hide tabline completely using `showtabline=0`

**Comment by mike-lloyd03** [comment]
I added this after `require("lualine").setup(...)` and it seems to work as desired:

```lua
vim.cmd("set showtabline=1")
```

**Comment by yetyetanotherusername** [comment]
For anyone hitting this issue in 2026, add `always_show_tabline = false` to your lualine options. This makes the brittle workaround with registering an autocommand on `OptionSet showtabline` unnecessary, as lualine then sets `showtabline = 1` by itself.

ie

```
options = {
    always_show_tabline = false,
}
```

Big thanks to whoever implemented this.
